jsinterop-ts-defs-processor/src/main/java/com/vertispan/tsdefs/model/TsProperty.java

```diff
@@ -63,6 +64,19 @@ public String emit(String indent, String ending, String parentNamespace) {
     return sb.toString();
   }
 
+  @Override
+  public boolean equals(Object o) {
+    if (this == o) return true;
+    if (!(o instanceof TsProperty)) return false;
+    TsProperty that = (TsProperty) o;
+    return Objects.equals(name, that.name);
```

Comment:
shouldnt we check the type also? or are we using equals to mean "this property hides that one"?

Reply:
Both java and typescript will give an error if 2 fields has the same name but different types. if we add this check then we are just adding it to guard against `@JsProperty` and in that case we will be throwing an error not just merge the properties.

So my conclusion a check for the name and type altogether should as a validation check somewhere else.

Reply:
Hmm, i'm curious about that, because you can definitely override methods and change types in TS...

yep, it works:

```typescript
interface Foo {
	get name():string
}
interface Bar extends Foo {
	get age():number
}

interface Accessor {
	mylist: Array<Foo>
}

interface Accessor2 extends Accessor {
	mylist: Array<Bar>
}
```
https://www.typescriptlang.org/play?filetype=d.ts#code/JYOwLgpgTgZghgYwgAgGIHt3IN4FgBQAkAOYRjIhwC2EAFAJQBcAzmFKMQQL4GiSyIUAIThRkEAB6QQAE2ZpMOAiTLI4pBoxABXKgCNo3Ar3DR4SZAEEESZs3Ri8RKgE8ANsFaMrUKHBcAPBjoAHxG+Cb85ijWtvZQAEziUhCy8rEQdg5Kzu6eYN6Wvv4BIlBh+FxAA

Reply:
type added to the check